client/app/scripts/reducers/root.js

```diff
@@ -741,6 +742,23 @@ export function rootReducer(state = initialState, action) {
       return state.set('nodesLoaded', false);
     }
 
+    case ActionTypes.REQUEST_SERVICE_IMAGES: {
+      return state.setIn(['serviceImages', 'fetching'], true);
+    }
+
+    case ActionTypes.RECEIVE_SERVICE_IMAGES: {
+      state = state.setIn(['serviceImages', 'fetching'], false);
+
+      if (action.errors) {
+        return state.setIn(['serviceImages', 'errors'], action.errors);
+      }
+
+      const service = find(action.services, s => s.ID === action.serviceId);
```

Comment:
In the spirit of keeping the reducers thin and straightforward, I'd just pick out and forward the service object from `getImagesForService` action instead of here. Then (together with the suggestion above), this whole action would simply be:

```
return state.mergeIn(['service', 'images', action.service.ID], {
  isFetching: false,
  errors: action.errors,
  data: action.service.Containers
});
```

Just an idea, up to you to decide!